client/src/components/pages/EditSurvey.js

```diff
@@ -0,0 +1,83 @@
+import React, { useState, useEffect } from "react";
+import { useParams } from "react-router-dom";
+
+import { Page, Spinner, Stack } from "@shopify/polaris";
+import { EditSurveyForm } from "..";
+
+export function EditSurvey() {
+  let { id: surveyId } = useParams();
+
+  const [name, setName] = useState("");
+  const [questions, setQuestions] = useState([]);
+
+  // a curry function that we will pass down to EditSurveyForm to update state
+  // This will be our state updated for all our survey files
+  // index will be the index of the question in in survey.questions,
+  // extraOpts is for updating mins/max in range questions or options in option questions
+  function handleQuestionChange(index, extraOpts) {
+    return newValue => {
+      const updatedQuestions = questions.map((question, idx) => {
```

Comment:
nit: I find the map misleading for the intention of this function. I feel like changing the value you intend to update is much more clear, something like this
```js
let updatedQuestions = [...questions];
if (extraOpts) {
  updatedQuestions[index] = {
    ...question,
    ...(extraOpts === "min" ? { min: newValue } : {}),
    ...(extraOpts === "max" ? { max: newValue } : {}),
    ...(extraOpts === "options" ? { options: newValue.split(",") } : {})
} else {
  updatedQuestions[index] = {
    ...question,
    question: newValue
  };
}
setQuestions(updatedQuestions); 

```